python/tskit/trees.py

```diff
@@ -7159,6 +7161,7 @@ def diversity(
         :param bool span_normalise: Whether to divide the result by the span of the
             window (defaults to True).
         :return: A numpy array whose length is equal to the number of sample sets.
+            If there is just one sample set and one window, a numpy scalar is returned.
```

Comment:
The reason why I updated these is because the output is no longer always a NumPy array now. What about something like
"Please see :ref:sec_stats_output_format for information about the shape and type of the returned NumPy object"?
Although I can see that the existing output is quite repetitive (and a pain to update), I think there's a case to make for leaving it in its current explicit form s well. "Dimensionality" can be quite an off-putting word to non-mathy folks, and I think talking about "dropping dimensions" is likely to be even more confusing as that is something that happens entirely under the hood here.